optimum/exporters/onnx/base.py

```diff
@@ -902,8 +903,8 @@ def post_process_exported_models(
             path, models_and_onnx_configs, onnx_files_subpaths
         )
 
-        # Attempt to merge only if the decoder was exported without/with past
-        if self.use_past is True and len(models_and_onnx_configs) == 3:
+        # Attempt to merge only if the decoder was exported without/with past, and ignore seq2seq models exported with text-generation task
+        if len(onnx_files_subpaths) >= 3 and self.use_past is True or self.variant == "with-past":
```

Comment:
why do we need to check  `self.variant` ?

Reply:
I'm not sure. I'll need to double check.